x/poe/contract/contractio_integration_test.go

```diff
@@ -35,44 +39,147 @@ func TestSetEngagementPoints(t *testing.T) {
+
+			// then
+			if spec.expErr {
+				require.Error(t, gotErr)
+				return
+			}
+			require.NoError(t, gotErr)
+
+			gotRes, err := contract.QueryStakedAmount(ctx, example.TWasmKeeper, stakingContractAddr, myOperatorAddr)
+			require.NoError(t, err)
+			expAmount := vals[0].Tokens.Add(spec.liquid.AmountOf("utgd")).String()
+			assert.Equal(t, expAmount, gotRes.Liquid.Amount)
+			assert.Equal(t, "utgd", gotRes.Liquid.Denom)
+
+			expAmount = "0"
+			if spec.vesting != nil {
+				expAmount = spec.vesting.Amount.String()
+			}
+			assert.Equal(t, expAmount, gotRes.Vesting.Amount)
+			assert.Equal(t, "utgd", gotRes.Vesting.Denom)
+		})
+	}
 
-	gotRes, err := contract.QueryStakedAmount(ctx, example.TWasmKeeper, stakingContractAddr, myOperatorAddr)
-	require.NoError(t, err)
-	expAmount := vals[0].Tokens.Add(sdk.OneInt())
-	assert.Equal(t, gotRes.Liquid.Amount, expAmount.String())
 }
 
 func TestUnbondDelegation(t *testing.T) {
 	// setup contracts and seed some data
 	ctx, example, vals, _ := setupPoEContracts(t)
+	stakingContractAddr, err := example.PoEKeeper.GetPoEContractAddress(ctx, types.PoEContractTypeStaking)
+	require.NoError(t, err)
 
 	myOperatorAddr, _ := sdk.AccAddressFromBech32(vals[0].OperatorAddress)
-	stakingContractAddr, err := example.PoEKeeper.GetPoEContractAddress(ctx, types.PoEContractTypeStaking)
+	genTxStakedAmount := vals[0].Tokens
+
+	vestingAmount := sdk.NewCoin(types.DefaultBondDenom, sdk.OneInt())
+	example.Faucet.Fund(ctx, myOperatorAddr, vestingAmount)
+	convertToFullVestingAccount(t, example, ctx, myOperatorAddr)
+
+	// and bond from vesting amount
+	err = contract.BondDelegation(ctx, stakingContractAddr, myOperatorAddr, sdk.NewCoins(), &sdk.Coin{Denom: types.DefaultBondDenom, Amount: sdk.OneInt()}, example.TWasmKeeper.GetContractKeeper())
 	require.NoError(t, err)
 	unbodingPeriod, err := example.PoEKeeper.StakeContract(ctx).QueryStakingUnbondingPeriod(ctx)
 	require.NoError(t, err)
 
-	// when
-	completionTime, err := contract.UnbondDelegation(ctx, stakingContractAddr, myOperatorAddr, sdk.NewCoin(types.DefaultBondDenom, sdk.OneInt()), example.TWasmKeeper.GetContractKeeper())
-
-	// then
-	require.NoError(t, err)
-	assert.Equal(t, ctx.BlockTime().Add(unbodingPeriod).UTC(), *completionTime)
+	parentCtx := ctx.WithBlockHeight(ctx.BlockHeight() + 1)
+	specs := map[string]struct {
+		unbondAmount   sdk.Coin
+		expLiquidBond  sdk.Int
+		expVestingBond sdk.Int
+		expErr         bool
+	}{
+		"some liquid": {
+			unbondAmount:   sdk.NewCoin(types.DefaultBondDenom, sdk.OneInt()),
+			expLiquidBond:  genTxStakedAmount.Sub(sdk.OneInt()),
+			expVestingBond: sdk.OneInt(),
+		},
+		"all liquid": {
+			unbondAmount:   sdk.NewCoin(types.DefaultBondDenom, genTxStakedAmount),
+			expLiquidBond:  sdk.ZeroInt(),
+			expVestingBond: sdk.OneInt(),
+		},
+		"all liquid + vesting": {
+			unbondAmount:   sdk.NewCoin(types.DefaultBondDenom, genTxStakedAmount.AddRaw(1)),
```

Comment:
Nit: It would be better to create a `Int` var for the vesting amount, and use it here and elsewhere.

Reply:
That would work. My rule of thumbs with tests is to have them self describing. `sdk.OneInt()` is in this case more self explaining than a `oneInt` var which could contain some other value or may be modified by tests.